src/Microsoft.Data.Analysis/ArrowStringDataFrameColumnRollingWindow.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Xml.Schema;
+
+namespace Microsoft.Data.Analysis
+{
+    public class ArrowStringDataFrameColumnRollingWindow : DataFrameColumnWindow
+    {
+        private int _windowSize;
+        private ArrowStringDataFrameColumn _currentColumn;
+        private long _currentIndex;
+
+        internal ArrowStringDataFrameColumnRollingWindow(int windowSize, ArrowStringDataFrameColumn currentColumn)
+        {
+            _windowSize = windowSize;
+            _currentColumn = currentColumn;
+        }
+
+        public override PrimitiveDataFrameColumn<int> Count() => StringDataFrameColumnRollingWindow.CountImplementation(_currentIndex, _currentColumn, _windowSize);
+
+        public override DataFrameColumn Max() => throw new NotSupportedException();
+
+        public override DataFrameColumn Min() => throw new NotSupportedException();
+
+        // ArrowStringDataFrameColumnRollingWindow does not get an Apply function because ArrowStringDataFrameColumn is immutable
```

Comment:
I'm not following this. The `Apply` function doesn't mutate the current column. But instead it returns a new instance of the column, right? We should be able to create a new instance given the values, or even create a `StringDataFrameColumn`. Maybe this could be logged as a separate issue, but this should be possible.

Reply:
Fixed now, this was my bad. It does get an `Apply` function. I pushed a half complete PR so I could pick it up on another machine. Should have marked it as WIP. This PR is ready for review now